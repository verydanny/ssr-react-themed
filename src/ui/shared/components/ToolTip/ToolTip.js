import React from 'react'
import themed from 'react-themed'
import PropTypes from 'prop-types'
import { Icon } from 'ui/shared/components/Icon'
import { close } from 'ui/shared/components/Icon/svgs/global'

const ToolTip = ({ theme, onClick, children }) => (

  <React.Fragment>
    <div
      role="button"
      tabIndex={0}
      className={theme.ToolTip_Overlay}
      onClick={onClick}
    />
    <div className={theme.ToolTip}>
      <div
        role="button"
        tabIndex={0}
        className={theme.ToolTip_Body}
        onClick={onClick}
      >
        <div
          role="button"
          tabIndex={0}
          className={theme.ToolTip_CloseButton}
          onClick={onClick}
        >
          <Icon svgs={{ close }} />
        </div>
        {children}
      </div>
    </div>
  </React.Fragment>
)

ToolTip.propTypes = {
  theme: PropTypes.object,
  onClick: PropTypes.func,
  children: PropTypes.node,
}

ToolTip.defaultProps = {
  theme: {},
}

export default themed(/^ToolTip/)(ToolTip)
