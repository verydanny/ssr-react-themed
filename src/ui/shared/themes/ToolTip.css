@value ( UniversalGrey ) from 'ui/shared/themes/colors/colors.css';

.ToolTip_Body {
  position: relative;
  background-color: white;
  filter: drop-shadow(0 0 0.75rem rgba(0,0,0,0.2));
  color: UniversalGrey;
  border-radius: 5px;
  z-index: 3;
}

.ToolTip_CloseButton {
  padding: 5px;
  position: absolute;
  top: 0;
  right:0;
  font-size: 28px;
}

/* NOTE: adds arrow to top of modal */
.ToolTip_Body::before {
  content: "";
  position: absolute;
  bottom: 100%;
  left: 50%;
  border-width: 10px;
  border-style: solid;
  border-color:  transparent  transparent #ffffff transparent;
}

.ToolTip_Overlay {
  position: fixed;
  width: 100%;
  height: 100%;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background-color: transparent;
  z-index: 2;
}
