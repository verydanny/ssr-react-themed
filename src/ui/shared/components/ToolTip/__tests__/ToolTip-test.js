import React from 'react'
import { shallow, mount } from 'enzyme'
import { keyMirror } from '@rentpath/react-ui-utils'
import ToolTip from '../ToolTip'

const theme = keyMirror([
  'ToolTip_CloseButton',
  'ToolTip_Overlay',
  'ToolTip_Body',
])

const defaultProps = {
  theme,
}

const factory = props =>
  <ToolTip.WrappedComponent {...defaultProps} {...props} />

describe('<ToolTip />', () => {
  it('renders', () => {
    const wrapper = shallow(factory())
    expect(wrapper).toMatchSnapshot()
  })

  it('renders children', () => {
    const children = <div className="Child" />
    const wrapper = mount(
      <ToolTip.WrappedComponent {...defaultProps}>
        {children}
      </ToolTip.WrappedComponent>
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('triggers an onClick for each button', () => {
    const onClick = jest.fn()
    const wrapper = shallow(factory({ onClick }))
    const selectors = ['.ToolTip_Body', '.ToolTip_Overlay', '.ToolTip_CloseButton']
    selectors.forEach(sel => wrapper.find(sel).simulate('click'))
    expect(onClick.mock.calls.length).toBe(3)
  })
})
