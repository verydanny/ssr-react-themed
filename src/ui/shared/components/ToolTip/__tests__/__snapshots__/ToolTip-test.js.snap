// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToolTip /> renders 1`] = `
<React.Fragment>
  <div
    className="ToolTip_Overlay"
    role="button"
    tabIndex={0}
  />
  <div>
    <div
      className="ToolTip_Body"
      role="button"
      tabIndex={0}
    >
      <div
        className="ToolTip_CloseButton"
        role="button"
        tabIndex={0}
      >
        <Themed(Icon)
          svgs={
            Object {
              "close": "",
            }
          }
        />
      </div>
    </div>
  </div>
</React.Fragment>
`;

exports[`<ToolTip /> renders children 1`] = `
<ToolTip
  theme={
    Object {
      "ToolTip_Body": "ToolTip_Body",
      "ToolTip_CloseButton": "ToolTip_CloseButton",
      "ToolTip_Overlay": "ToolTip_Overlay",
    }
  }
>
  <div
    className="ToolTip_Overlay"
    role="button"
    tabIndex={0}
  />
  <div>
    <div
      className="ToolTip_Body"
      role="button"
      tabIndex={0}
    >
      <div
        className="ToolTip_CloseButton"
        role="button"
        tabIndex={0}
      >
        <Themed(Icon)
          svgs={
            Object {
              "close": "",
            }
          }
        >
          <Icon
            svgs={
              Object {
                "close": "",
              }
            }
            theme={Object {}}
          />
        </Themed(Icon)>
      </div>
      <div
        className="Child"
      />
    </div>
  </div>
</ToolTip>
`;
